### ISSUE TITLE
ServiceContext->UsePlatformServices results in a blank baseserviceURL in $this->context

### ISSUE THREAD

**Issue by muggy8** [intuit/QuickBooks-V3-PHP-SDK#28]
# Edited: 

I believe I have found the source of the problem. Please see comment below for what I think is the issue. 

# context: 

I'm not sure if this is me doing something wrong or if the SDK is not working. However I am trying to implement a disconnect from QBO button in the app that I'm working on and after setting up the following 

```php
// my other code that contains my keys and stuff above ^^^

$dataServiceConfig = [
    'auth_mode' => 'oauth1',
    'consumerKey' => OAUTH_CONSUMER_KEY,
    'consumerSecret' => OAUTH_CONSUMER_SECRET,
    'accessTokenKey' => $authToken,
    'accessTokenSecret' => $authTokenSecret,
    'QBORealmID' => $realmId,
    'baseUrl' => "https://sandbox-quickbooks.api.intuit.com/"
];

$serviceContext = ServiceContext::ConfigureFromPassedArray($dataServiceConfig);

$platformService = new PlatformService($serviceContext);

// everything from this point on fails and outputs NULL
$Respxml = $platformService->CurrentUser();
var_dump($Respxml);

$Respxml = $platformService->Disconnect();
var_dump($Respxml);
```

I'm not sure if this is because what i've done and how i've initialized the instance of PlatformService is incorrect or if the PlatformService class itself is broken. so i looked into the logs to see what was happening and this is what i got out of them

```
RESPONSE URI FOR SEQUENCE ID 01497
==================================
v1/Connection/Disconnect

RESPONSE HEADERS
================


RESPONSE BODY
=============


```

```RESPONSE URI FOR SEQUENCE ID 01496
==================================
v1/user/current

RESPONSE HEADERS
================


RESPONSE BODY
=============


```

a typical response looks like this for me

```
RESPONSE URI FOR SEQUENCE ID 01485
==================================
https://sandbox-quickbooks.api.intuit.com/v3/company/__ID_hidden__/query?minorversion=4

RESPONSE HEADERS
================
Server: nginx
Content-Type: application/xml;charset=UTF-8
... some other stuff i've decided to hide bla bla

RESPONSE BODY
=============
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>... the rest of the response goes here


```

the typical response has a DataService object initiated in this way 

```php
$dataServiceConfig = [
    'auth_mode' => 'oauth1',
    'consumerKey' => OAUTH_CONSUMER_KEY,
    'consumerSecret' => OAUTH_CONSUMER_SECRET,
    'accessTokenKey' => $authToken,
    'accessTokenSecret' => $authTokenSecret,
    'QBORealmID' => $realmId,
    'baseUrl' => "https://sandbox-quickbooks.api.intuit.com/"
];

$dataService = DataService::Configure($dataServiceConfig);
```

Thanks for the help!

**Comment by muggy8** [comment]
For anyone out there who's looking. I have dug through the code of the SDK and i believe the source of the problem is in the ServiceContext class. 

```php
//...other code

    /**
     * Gets the base Uri for a QBO user.
     *
     * @return string Returns the base Uri endpoint for a user.
     */
    public function getBaseURL()
    {
        $this->IppConfiguration->Logger->RequestLog->Log(TraceLevel::Info, "Called GetBaseURL method.");
        try {
            if ($this->serviceType === CoreConstants::IntuitServicesTypeQBO) {
                $baseurl = $this->IppConfiguration->BaseUrl->Qbo . implode(CoreConstants::SLASH_CHAR, array(CoreConstants::VERSION)) . CoreConstants::SLASH_CHAR;
            }
            else {
                $baseurl = $this->IppConfiguration->BaseUrl->Ipp;
            }
        } catch (\Exception $e) {
            throw new \Exception("Base URL is not setup");
        }
        return $baseurl;
    }

    /**
     * Populates the values of the service context like service type and base url to Platform Services.
     */
    public function UsePlatformServices()
    {
        $this->serviceType = CoreConstants::IntuitServicesTypeIPP;
        $this->baseserviceURL = $this->getBaseURL();
    }

//... other code```

when i tried adding a var_dump($this->baseserviceURL) at the end of `public function UsePlatformServices()` it outputted a null which then results in a failed URL in the `GetResponse()` function of SyncRequestHandler class. and the problem seems to be t hat there's no action in the `getBaseURL()` function that handles `$this->serviceType == CoreConstants::IntuitServicesTypeIPP` Hope this helps someone out there figure out how to resolve this issue!

**Comment by renedekat** [comment]
This is exactly the issue I just found as well

`$this->serviceType = CoreConstants::IntuitServicesTypeIPP;`

Which uses the IPP baseUrl, and that hasn't been set.
The Disconnect method fails, but also the Reconnect which makes it impossible to refresh auth tokens.

**Comment by mjhca** [comment]
Yes.  This is a duplicate of issue #24.  It is a problem when configuration is done from an array rather than a file.  

Temporary solution is to call 
`$serviceContext->IppConfiguration->BaseUrl->Ipp = "https://appcenter.intuit.com/api/";`
after creating PlatformService but before trying to use it.

**Comment by hlu2** [comment]
@all

I think there is some issue with the Platform service call. Let me take a look and see what the problem is.

**Comment by gravit8** [comment]
I also had this issue, when calling  $platformService->Disconnect(); I was getting a false Boolean. I thought that I had the wrong endpoint for a while, @muggy8 fix worked for me thanks.

**Comment by hlu2** [comment]
@ALL

This should be fixed on v3.2.6.
All Platform calls should work properly now.
Any issue, ping me.
